neo/SmartContract/ApplicationEngine.cs

```diff
@@ -228,6 +229,18 @@ private bool CheckStackSize(OpCode nextInstruction)
             return true;
         }
 
+        private bool CheckDynamicInvoke(OpCode nextInstruction)
+        {
+            if(nextInstruction == OpCode.DYNAMICCALL)
+            {
+                byte[] script_hash = EvaluationStack.Peek().GetByteArray();
+                CachedScriptTable script_table = (CachedScriptTable)this.table;
+                ContractState contract_state = script_table.GetContractState(script_hash);
+                return contract_state.HasDynamicInvoke;
```

Comment:
Here you obtain `HasDynamicInvoke` from the first item of the stack, but in fee calculation at the third. 
If both are good, im prefer the third for save `CachedScriptTable` logics

Reply:
That is a good point.  Better to make both of them use `CachedScriptTable`

Reply:
what is the faster way?

Reply:
The fee calculation in the third would be faster by a lot I imagine

Reply:
@shargon I changed them to both use `EvaluationStack.Peek(3)` as it is faster.  This also allows me to not need to use `CachedScriptTable`, and keep the `IScriptTable table` property on `ExecutionEngine.cs` in the VM private instead of protected.